### PR TITLE
[fix] Adjust logic for applying outcome for IF_ELSE_CONDITION tasks

### PR DESCRIPTION
**Pull request by mikita-sakalouski**
## Description

1. Add check that outcome is applied on correct task from dependencies
2. Get correct outcome based on task key and not just first one

## Related Issue
#172 #171 

## Motivation and Context
Get correct behavior

## How Has This Been Tested?
Added more tasks into sample workflow for testing

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mikita-sakalouski** [comment]
Depends on #168 . Please firstly review and merge PR #168

File: brickflow/engine/task.py [hunk 1]
```diff
@@ -822,15 +822,24 @@ def parents(self) -> List[str]:
     @property
     def depends_on_names(self) -> Iterator[Dict[str, Optional[str]]]:
         for i in self.depends_on:
-            if self.if_else_outcome:
-                outcome = list(self.if_else_outcome.values())[0]
+            task_name = i.__name__ if callable(i) and hasattr(i, "__name__") else str(i)
```
Comment (newfront) [inline brickflow/engine/task.py:825]: you end up testing `if callable(i) and hasattri(i, "__name__")` twice per loop. 
You could store in a `task_is_callable`  variable and then use it like.

~~~python
task_is_callable = True if callable(i) and hasattri(i, "__name__") else False
task_name = i.__name__ if task_is_callable else str(i)
...
if task_is_callable:
  yield {task_name: outcome}
else:
  yield {task_name: outcome}
~~~

However, given you only really need to test once when you set `task_name` on line 825,
the final block could just be condensed to `yield { task_name: outcome }`

**Review by newfront — DISMISSED** [review]
@mikita-sakalouski There is a small fix that could simplify the block on `engine/task.py` - see my comment where we could remove the closing block since the `yield {task_name: outcome}` is the same on either side of the condition. 

Otherwise this looks good to ship. :shipit:

File: brickflow/engine/task.py (hunk 1 above)
Comment (mikita-sakalouski) [reply brickflow/engine/task.py:825]: Fixed, thanks!

**Review by pariksheet — APPROVED** [review]
LGTM

**Pull request merged**